### PR TITLE
2713kg7 softdelete test doc

### PR DESCRIPTION
**Pull request by shubham-shay-te**

### CODE REVIEW

File: core_lib/data_layers/data/db/sqlalchemy/mixins/soft_delete_mixin.py [hunk 1]
```diff
@@ -6,15 +6,7 @@
 class SoftDeleteMixin(object):
     created_at = Column(DateTime, default=datetime.utcnow)
     created_at._creation_order = 9998
-    updated_at = Column(DateTime, default=datetime.utcnow)
+    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
```
Comment (shay-te) [inline core_lib/data_layers/data/db/sqlalchemy/mixins/soft_delete_mixin.py:9]: Thank you this looks much more cleaner. 

please can you share a link to read where `__declare_last__` will not happen all the time.

File: tests/test_soft_delete.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    def setUpClass(cls):
+        cls.db_data_session = connect_to_mem_db()
+
+    def test_soft_delete(self):
+        data_name_1 = "test_name_1"
+        data_name_2 = "test_name_2"
+        data_name_3 = "test_name_3"
+        with self.__class__.db_data_session.get() as session:
+            data = Data()
+            data.name = data_name_1
```
Comment (shay-te) [inline tests/test_soft_delete.py:34]: you can use the same session to add all 3 Data objects.

File: website/docs/soft_delete.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    created_at._creation_order = 9998
+    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
+    updated_at._creation_order = 9998
+    deleted_at = Column(DateTime, default=None)
+    deleted_at._creation_order = 9998
+```
+
+
+### SoftDeleteTokenMixin
+`SoftDeleteTokenMixin` creates the following column in a table. Which can be used for Indexing if the user wants.
```
Comment (shay-te) [inline website/docs/soft_delete.md:41]: forgot to mention that. 
When index is needed on deleted_at column. it's better to user on the delete_token for reason that index on a datetime column are very slow. 

This is the only reason for this class to exists,. so it's worth mentioning that

File: core_lib/data_layers/data/db/sqlalchemy/mixins/soft_delete_mixin.py (hunk 1 above)
Comment (shubham-shay-te) [reply core_lib/data_layers/data/db/sqlalchemy/mixins/soft_delete_mixin.py:9]: https://docs.sqlalchemy.org/en/14/orm/events.html#sqlalchemy.orm.MapperEvents.before_update

Comment (shay-te) [reply core_lib/data_layers/data/db/sqlalchemy/mixins/soft_delete_mixin.py:9]: thank you and where dose it say the `onupdate` is guaranteed to call?

Comment (shubham-shay-te) [reply core_lib/data_layers/data/db/sqlalchemy/mixins/soft_delete_mixin.py:9]: https://docs.sqlalchemy.org/en/14/core/metadata.html?highlight=onupdate#sqlalchemy.schema.Column.params.onupdate

**Review by shay-te — APPROVED** [review]

**Pull request merged**